src/mds/Server.cc

```diff
@@ -1046,6 +1102,93 @@ void Server::respond_to_request(MDRequestRef& mdr, int r)
     mdr->internal_op_finish->complete(r);
     mdcache->request_finish(mdr);
   }
+
+  // add here to avoid counting ops multiple times (e.g., locks, loading)
+  switch(mdr->client_request->get_op()) {
```

Comment:
That massive switch statement has internal ops (like snapshot and file layout stuff), right? So we want a conditional like:

```C++
if (mdr->client_request || mdr->internal_op > -1) {
```

I suppose I could split the perf counters up -- but I'm not sure which ones are internal ops.